### PR TITLE
chore(lib/cmd): improve invalid command errors

### PR DESCRIPTION
**Pull request by corverroos**
Improve binary errors when subcommands are invalid or not specified, reverting to default cobra behaviour.

issue: none

### CODE REVIEW

**Comment by corverroos** [comment]
This will avoid the following logs during cosmovisor restarts which are misleading.
image

File: lib/cmd/cmd.go [hunk 1]
```diff
@@ -20,32 +20,27 @@ import (
...
-	SilenceErrUsage(cmd)
+	wrapRunCmd(cmd)
 
+	ctx, cancel := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
 	err := cmd.ExecuteContext(ctx)
-
 	cancel()
 
 	if err != nil {
-		log.Error(ctx, "!! Fatal error occurred, app died unexpectedly !!", err)
```
Comment (corverroos) [inline lib/cmd/cmd.go:39]: Also slightly improves fatal errors logs since since we now have "expected fatal errors"

**Comment by corverroos** [comment]
This needs to be backported to `upgrade/v0.8`

File: lib/cmd/cmd.go [hunk 2]
```diff
@@ -168,3 +167,30 @@ func bindFlags(cmd *cobra.Command, v *viper.Viper) error {
 
 	return lastErr
 }
+
+// wrapRunCmd wraps the "app run" command to custom fatal error log and silence cobra output.
+func wrapRunCmd(cmd *cobra.Command) {
+	runCmd := getRunCmd(cmd)
+	SilenceErrUsage(runCmd)
+	runFunc := runCmd.RunE
```
Comment (sideninja) [inline lib/cmd/cmd.go:175]: lacking a bit of context but is it safe to assume RunE won't be nil?

**Review by sideninja — APPROVED** [review]

File: lib/cmd/cmd.go (hunk 2 above)
Comment (corverroos) [reply lib/cmd/cmd.go:175]: it would be a misconfiguration be us, but sure will add a check

**Comment by corverroos** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|release/v0.8|[](https://github.com/omni-network/omni/pull/2057)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

**Pull request merged**